### ISSUE TITLE
An error occurred while Performance/RegexpMatch cop

### ISSUE THREAD

**Issue by deepj** [bbatsov/rubocop#3911]
After the upgrade on RuboCop 0.47.0, run `rubocop` command, the following issue is triggered:

```
An error occurred while Performance/RegexpMatch cop
```

That happens on this code:

```ruby
if head =~ /charset=([^\s"']+)/i
```

`rubocop -V`:

```
0.47.0 (using Parser 2.3.3.1, running on ruby 2.3.3 x86_64-darwin16)
```

**Comment by pocke** [comment]
Thanks for your feedback!

I can't reproduce this problem.
Could you give me more detailed information and stacktrace with `-d` option?

**Comment by deepj** [comment]
Here is the whole method where it's failing
```ruby
def self.detect_charset_from_html(raw_html)
  head = raw_html[/<head>.*<\/head>/im]
  if head =~ /charset=([^\s"']+)/i
    $1
  else
    nil
  end
end
```

Stacktrace:

```
An error occurred while Performance/RegexpMatch cop was inspecting /content_utils.rb:271:6.
undefined method `each_node' for nil:NilClass
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/performance/regexp_match.rb:100:in `search_match_nodes'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/performance/regexp_match.rb:159:in `each'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/performance/regexp_match.rb:159:in `find'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/performance/regexp_match.rb:159:in `next_match_pos'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/performance/regexp_match.rb:152:in `last_match_used?'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/performance/regexp_match.rb:141:in `block in check_condition'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/performance/regexp_match.rb:98:in `match_node?'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/performance/regexp_match.rb:140:in `check_condition'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/performance/regexp_match.rb:110:in `on_if'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:43:in `block (2 levels) in on_if'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:102:in `with_cop_error_handling'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:42:in `block in on_if'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:41:in `each'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:41:in `on_if'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:47:in `on_begin'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/ast/traversal.rb:122:in `on_defs'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:47:in `on_defs'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:47:in `on_begin'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/ast/traversal.rb:141:in `on_while'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:47:in `on_module'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/ast/traversal.rb:141:in `on_while'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:47:in `on_module'
(eval):2:in `block in on_begin'
(eval):2:in `each'
(eval):2:in `on_begin'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:47:in `on_begin'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/ast/traversal.rb:12:in `walk'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/commissioner.rb:60:in `investigate'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/team.rb:121:in `investigate'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/team.rb:109:in `offenses'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cop/team.rb:51:in `inspect_file'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:248:in `inspect_file'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:195:in `block in do_inspection_loop'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:227:in `block in iterate_until_no_changes'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:220:in `loop'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:220:in `iterate_until_no_changes'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:191:in `do_inspection_loop'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:101:in `block in file_offenses'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:111:in `file_offense_cache'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:99:in `file_offenses'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:90:in `process_file'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:68:in `block in each_inspected_file'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:65:in `each'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:65:in `reduce'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:65:in `each_inspected_file'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:57:in `inspect_files'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/runner.rb:36:in `run'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cli.rb:72:in `execute_runner'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/lib/rubocop/cli.rb:27:in `run'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/bin/rubocop:13:in `block in <top (required)>'
~/.rubies/ruby-2.3.3/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
~/.gem/ruby/2.3.3/gems/rubocop-0.47.0/bin/rubocop:12:in `<top (required)>'
~/.gem/ruby/2.3.3/bin/rubocop:22:in `load'
~/.gem/ruby/2.3.3/bin/rubocop:22:in `<main>'
```
